src/mixins/name-request-mixin.ts

```diff
@@ -112,6 +116,21 @@ export default class NameRequestMixin extends Mixins(DateMixin) {
     if (nr.state === NameRequestStates.CONDITIONAL) {
       return nr.names.find(name => name.state === NameRequestStates.CONDITION).name
     }
-    return '' // should never happen
+    return null // should never happen
+  }
+
+  /**
+   * Returns the entity (corp) type that corresponds to the NR type.
+   * @param nr the name request response payload
+   */
+  nrTypeToEntityType (nr: any): CorpTypeCd {
+    switch (nr.entity_type_cd) {
+      case NameRequestTypes.BC: return CorpTypeCd.BENEFIT_COMPANY
+      case NameRequestTypes.CR: return CorpTypeCd.BC_CORPORATION
+      case NameRequestTypes.UL: return CorpTypeCd.BC_UNLIMITED
+      case NameRequestTypes.CC: return CorpTypeCd.BC_CCC
+      case NameRequestTypes.CP: return CorpTypeCd.COOP
+    }
+    return null // unknown entity type code
```

Comment:
This seems like a trivial method, but it does handle BC -> BEN, and it also returns null for NR entity type codes we don't support yet.